tests/test_flaggedSourceSelector.py

```diff
@@ -0,0 +1,123 @@
+    values to distinguish them in catalogs with multiple objects.
+    """
+    record = src.addNew()
+    record['coord_ra'] = (1. + num) * lsst.afw.geom.degrees
+    record['coord_dec'] = (2. + num) * lsst.afw.geom.degrees
+    record['slot_Centroid_x'] = 10. + num
+    record['slot_Centroid_y'] = 20. + num
+    record['slot_ApFlux_flux'] = 100. + num
+    record['slot_ApFlux_fluxSigma'] = 1.
+    record.set("calib_psfUsed", True)
+
+
+class TestFlaggedSourceSelector(lsst.utils.tests.TestCase):
+
+    def setUp(self):
+        schema = lsst.meas.base.tests.TestDataset.makeMinimalSchema()
+        schema.addField("slot_ApFlux_flux", type=float)
+        schema.addField("slot_ApFlux_fluxSigma", type=float)
+        schema.addField("calib_psfUsed", type="Flag")
+        schema.addField("is_selected", type="Flag")
+
+        self.selectedKey = "is_selected"
+
+        self.src = afwTable.SourceCatalog(schema)
+        self.sourceSelector = sourceSelector.sourceSelectorRegistry['flagged']()
+
+    def tearDown(self):
+        del self.src
+        del self.sourceSelector
+
+    def testSelectSourcesGood(self):
+        """Insert good sources and check that they were selected."""
+        for i in range(5):
+            addGoodSource(self.src, i)
+        result = self.sourceSelector.run(self.src)
+        for x in self.src['id']:
+            self.assertIn(x, result.sourceCat['id'])
+
+    def testSelectSourcesSelectedField(self):
+        """Test the behavior of source_selected_field in run.
+
+        This test asserts that the field will specified in
+        ``source_selected_field`` is properly set by the source selector.
+        We test this both for sources that fail and pass our cuts.
+        """
+        for i in range(5):
+            addGoodSource(self.src, i)
+        self.src[0].set("calib_psfUsed", False)
+        self.sourceSelector.run(self.src, sourceSelectedField=self.selectedKey)
+        for src_idx in range(5):
+            if src_idx == 0:
+                self.assertFalse(self.src[src_idx].get("is_selected"))
+            else:
+                self.assertTrue(self.src[src_idx].get("is_selected"))
+
+    def testSelectSourcesBad(self):
+        """Add a source that fails the source selector test and check
+        that our output array is indeed empty.
+        """
+        addGoodSource(self.src, 1)
+        self.src[0].set('calib_psfUsed', False)
+        result = self.sourceSelector.run(self.src)
+        self.assertNotIn(self.src['id'][0], result.sourceCat['id'])
+
+    def testSelectSourcesNonContiguous(self):
+        """Cannot do source selection on non-contiguous catalogs."""
+        for i in range(3):
+            addGoodSource(self.src, i)
+        del self.src[1]  # take one out of the middle to make it non-contiguous.
+        self.assertFalse(self.src.isContiguous(), "Catalog is contiguous: the test won't work.")
+
+        with self.assertRaises(RuntimeError):
+            self.sourceSelector.run(self.src)
```

Comment:
Fair point. I added that explicit test because the defined API says an exception is raised. In fact, this reminds me that I should make that `raise` explicit in the docstring.

I'll pull the test out of the flagged and astrometry testers and put it in a trivial BaseSourceSelectorTestCase, which is where it belongs.

RFC-335 said "...raise the exception afw table raises when one attempts to run vector operations on a non-contiguous catalog in Python," which is `lsst.pex.exceptions.RuntimeError`, so `RuntimeError` is the appropriate exception.